pywr/solvers/cython_glpk_edge.pyx

```diff
@@ -0,0 +1,742 @@
+                # change in storage cannot be more than the current volume or
+                # result in maximum volume being exceeded
+                lb = -avail_volume/timestep._days
+                ub = max(max_volume - storage._volume[scenario_index.global_id], 0.0) / timestep._days
+
+                if abs(lb) < 1e-8:
+                    lb = 0.0
+                if abs(ub) < 1e-8:
+                    ub = 0.0
+                set_row_bnds(self.prob, self.idx_row_virtual_storages+col, constraint_type(lb, ub), lb, ub)
+
+        self.stats['bounds_update_storage'] += time.perf_counter() - t0
+
+        t0 = time.perf_counter()
+
+        # Apply presolve if required
+        if self.use_presolve and not self.has_presolved:
+            self.smcp.presolve = GLP_ON
+            self.has_presolved = True
+        else:
+            self.smcp.presolve = GLP_OFF
+
+        # Set the basis for this scenario
+        self._set_basis(scenario_index.global_id)
+        # attempt to solve the linear programme
+        simplex_ret = simplex(self.prob, self.smcp)
+        status = glp_get_status(self.prob)
+        if (status != GLP_OPT or simplex_ret != 0) and self.retry_solve:
+            # try creating a new basis and resolving
+            print('Retrying solve with new basis.')
+            glp_std_basis(self.prob)
+            simplex_ret = simplex(self.prob, self.smcp)
+            status = glp_get_status(self.prob)
+
+        if status != GLP_OPT or simplex_ret != 0:
+            # If problem is not solved. Print some debugging information and error.
+            print("Simplex solve returned: {} ({})".format(simplex_status_string[simplex_ret], simplex_ret))
+            print("Simplex status: {} ({})".format(status_string[status], status))
+            print("Scenario ID: {}".format(scenario_index.global_id))
+            print("Timestep index: {}".format(timestep._index))
+            self.dump_mps(b'pywr_glpk_debug.mps')
+            self.dump_lp(b'pywr_glpk_debug.lp')
+
+            self.smcp.msg_lev = GLP_MSG_DBG
+            # Retry solve with debug messages
+            simplex_ret = simplex(self.prob, self.smcp)
+            status = glp_get_status(self.prob)
+            raise RuntimeError('Simplex solver failed with message: "{}", status: "{}".'.format(
+                simplex_status_string[simplex_ret], status_string[status]))
+        # Now save the basis
+        self._save_basis(scenario_index.global_id)
+
+        self.stats['lp_solve'] += time.perf_counter() - t0
+        t0 = time.perf_counter()
+
+        cdef double[:] edge_flows = self.edge_flows_arr
+
+        for col in range(self.num_edges):
+            edge_flows[col] = glp_get_col_prim(self.prob, col+1)
+
+        cdef double[:] change_in_storage = self.change_in_storage_arr
+        # for col in range(0, self.num_storages):
+        #     change_in_storage[col] = glp_get_row_prim(self.prob, self.idx_row_storages+col)
```

Comment:
If this is no longer used then it should be removed. This could be done in another PR/issue if it's a problem for the other solver.